src/functions_framework/__init__.py

```diff
@@ -65,6 +65,17 @@ def __init__(
         self.data = data
 
 
+class _LoggingHandler(object):
+    """Logging replacement for stdout and stderr in GCF Python 3.7."""
+
+    def __init__(self, level):
+        self.level = level
+
+    def write(self, out):
+        if out.rstrip() != "":
+            self.level(out.rstrip())
```

Comment:
```suggestion
        if out.rstrip():
            self.level(out.rstrip())
```

Also, I'm guessing this mirrors the existing behavior somehow? I would normally expect empty and whitespace-only strings to still be logged.

Reply:
This is because writing to sys.stdout/stderr with print() adds a newline, so it writes an additional empty log and confuses the Stackdriver parsing.

Reply:
I can run some more tests to see if the output is as expected.

Reply:
Got it, that makes sense.

Reply:
Modified the logging to not use the google-cloud-logging library, was having issues replicating the newlines that way. Empty and whitespace logs should work now.